src/compiler/checker.ts

```diff
@@ -274,6 +274,11 @@ namespace ts {
             getJsxNamespace: () => unescapeLeadingUnderscores(getJsxNamespace()),
             getAccessibleSymbolChain,
             resolveExternalModuleSymbol,
+
+            getTypeArgumentConstraint: node => {

```

Comment:
It feels like the name here is a perfect setup for someone accidentally misusing this API. I can definitely imagine someone calling this with a type parameter instead of a type argument, but I don't entirely know how to avoid that apart from something like `getTypeParameterConstraintForTypeArgument`.

As long as this is internal, I think this is fine though.

Reply:
Luckily `TypeParameter` isn't assignable to `TypeNode`, so trying to use it that way would fail.